src/Reveal.Sdk.Dom/Visualizations/Visualization.cs

```diff
@@ -106,6 +106,11 @@ public void UpdateDataSourceItem(DataSourceItem dataSourceItem)
             if (DataDefinition is TabularDataDefinition tdd)
             {
                 tdd.Fields = dataSourceItem.Fields.Clone();
+                
+                if(tdd.DataSourceItem.JoinTables != null)

```

Comment:
this can never be null because it's an internal property that has an initializer. To be super safe we can keep the null check, but we need to be checking if there are any items in the collection

Reply:
Added the count check